src/integTest/kotlin/io/snyk/SnykBulkFileListenerTest.kt

```diff
@@ -101,27 +100,95 @@ class SnykBulkFileListenerTest : BasePlatformTestCase() {
     fun testCurrentIacResults_shouldDropCachedResult_whenIacSupportedFileChanged() {
         val file = "k8s-deployment.yaml"
         val filePath = "/src/$file"
+        createFakeIacResultInCache(file, filePath)
+
+        myFixture.configureByText(file, "some text")
+
+        await().atMost(2, TimeUnit.SECONDS).until { cacheUpdated(filePath) }
+    }
+
+    private fun cacheUpdated(filePath: String?): Boolean {
+        val iacCachedIssues = project.service<SnykToolWindowPanel>().currentIacResult!!.allCliIssues!!
+        return iacCachedIssues.any { iacFile ->
+            (filePath == null || iacFile.targetFilePath == filePath) && iacFile.obsolete
```

Comment:
What is the purpose to check here for filePath == null? Should we ever have entries in iacCachedIssues when the filePath == null? In which scenario would that be a valid cache entry?

Reply:
filePath is method param we check against. 
`filePath == null` is the case when we want to check if _any_ IaC file with issues been marked as obsolete
adding kdoc to make it clear for the future (me) reader.